src/utils/agent_tools.py

```
@@ -923,22 +923,32 @@ async def create_observations(
+        try:
+            with embedding_call_purpose(
+                EmbeddingCallPurpose.CREATE_OBSERVATIONS.value,
+                workspace_name=workspace_name,
+                run_id=run_id,
+                parent_category=parent_category,
+            ):
+                embeddings = await embedding_client.simple_batch_embed(contents)
+            embeddings_by_index = dict(
+                zip(range(len(normalized_observations)), embeddings, strict=True)
+            )
+        except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)